src/MLIR/Native.hs

```diff
@@ -133,6 +135,15 @@ getUnknownLocation :: Context -> IO Location
 getUnknownLocation ctx =
   [C.exp| MlirLocation { mlirLocationUnknownGet($(MlirContext ctx)) } |]
 
+getFileLineColLocation :: Context -> StringRef -> C.CUInt -> C.CUInt -> IO Location
+getFileLineColLocation ctx (StringRef sPtr len) line col  =
+  [C.exp| MlirLocation {
+    mlirLocationFileLineColGet(
+      $(MlirContext ctx),
+      (MlirStringRef){$(char* sPtr), $(size_t len)},
```

Comment:
Quick question about ownership: does the newly created location expect that the string ref remains valid for its whole lifetime, or does it intern it somewhere in the MLIR context. Do locations have to be destroyed explicitly?

Reply:
It is interned & uniqued in the MLIR context yes. Currently they are only destroyed with the context.